File: nave-backend/src/main/java/reto/sofka2022/navebackend/modelos/Nave.java
package reto.sofka2022.navebackend.modelos;


//Componentes externos
import lombok.Data;
import lombok.Setter;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AccessLevel;

@Document(collection = "tb-Nave")
@Data


public class Nave {

    @Id
    @Setter(AccessLevel.NONE) //le establezco acces none para que no pueda modificarse el id
    
    private String _id;
    private String nombre;
    private String pais;
    private String tipoNave;
    
}

File: nave-frontEnd/src/app/modelo/nave.ts
export class Nave {

    _id: string;
    nombre: string;
    pais: string;
    tipoNave: string;

    constructor(_id: string, nombre: string, pais: string, tipoNave: string ){
        this._id = _id;
        this.nombre = nombre;
        this.pais = pais;
        this.tipoNave = tipoNave;
    }
}
